File: python/LeetCode/01758.py
def minOperations(self, s: str) -> int:
    # consider the following two cases
    # case 1 s starts with 0
    # case 2 s starts with 1

    case1 = 0
    case2 = 0
    for i in range(len(s)):
        if i % 2 == 0:
            if s[i] == '1':
                case1 += 1
            else:
                case2 += 1
        else:
            if s[i] == '1':
                case2 += 1
            else:
                case1 += 1

    return min(case1, case2)

File: go/leetcode/01758.go
package main

func minOperations(s string) int {
	c1 := 0
	c2 := 0

	for i, c := range s {
		if i%2 == 0 {
			if c == '1' {
				c1 += 1
			} else {
				c2 += 1
			}
		} else {
			if c == '1' {
				c2 += 1
			} else {
				c1 += 1
			}
		}
	}
	if c1 < c2 {
		return c1
	}
	return c2
}
